gemini/prompts/prompt_optimizer/vertex_ai_prompt_optimizer_ui.ipynb

```diff
@@ -0,0 +1,907 @@
+        "        with self.results_output:\n",
+        "            self.results_output.clear_output(wait=True)  # Clear previous output\n",
+        "            display_dataframe(self.templates[index])\n",
+        "            print()\n",
+        "            display_dataframe(self.eval_results[index])\n",
+        "\n",
+        "    def get_container(self):\n",
+        "        return widgets.VBox(\n",
+        "            [\n",
+        "                self.run_label,\n",
+        "                self.run_dropdrown,\n",
+        "                self.dropdown_description,\n",
+        "                self.template_dropdown,\n",
+        "                self.results_output,\n",
+        "            ]\n",
+        "        )"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "-p59jd5rOp4q"
+      },
+      "source": [
+        "# Step 1: Configure your prompt template\n",
+        "Prompts consist of two key parts:\n",
+        "* System Instruction (SI) Template: A fixed instruction shared across all queries for a given task.\n",
+        "* Task/Context Template: A dynamic part that changes based on the task.\n",
+        "\n",
+        "APD enables the translation and optimization of the System Instruction Template, while the Task/Context Template remains essential for evaluating different SI templates."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "rJG1pVZO317x"
+      },
+      "outputs": [],
+      "source": [
+        "INSTRUCTION_TEMPLATE = \"Answer the following question. Let's think step by step.\\n\"  # @param {type:\"string\"}\n",
+        "CONTEXT_TASK_TEMPLATE = (\n",
+        "    \"Question: {{question}}\\n\\nAnswer:{{target}}\"  # @param {type:\"string\"}\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5y-cmg0TQP6v"
+      },
+      "source": [
+        "# Step 2: Input your data\n",
+        "To optimize the model, provide a CSV or JSONL file containing labeled validation samples\n",
+        "* Focus on examples that specifically demonstrate the issues you want to address.\n",
+        "* Recommendation: Use 50-100 distinct samples for reliable results. However, the tool can still be effective with as few as 5 samples.\n",
+        "\n",
+        "For prompt translation:\n",
+        "* Consider using the source model to label examples that the target model struggles with, helping to identify areas for improvement.\n",
+        "\n",
+        "Learn more about setting up your CSV or JSONL file as input [TODO: add link]."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "mfgi_oR6tTIB"
+      },
+      "outputs": [],
+      "source": [
+        "# @markdown **Project setup**: <br/>\n",
+        "PROJECT_ID = \"vertex-research-solutions\"  # @param {type:\"string\"}\n",
```

Comment:
Replace this project ID with a placeholder.

Reply:
Done.